private/private.go

```diff
@@ -58,42 +58,31 @@ func FromEnvironmentOrNil(name string) PrivateTransactionManager {
 }
 
 func MustNewPrivateTxManager(cfgPath string) PrivateTransactionManager {
-	info, err := os.Lstat(cfgPath)
+	cfg, err := engine.LoadConfig(cfgPath)
 	if err != nil {
 		panic(fmt.Sprintf("unable to read %s due to %s", cfgPath, err))
 	}
-	// We accept either the socket or a configuration file that points to
-	// a socket.
-	socketPath := cfgPath
-	isSocket := info.Mode()&os.ModeSocket != 0
-	if !isSocket {
-		cfg, err := engine.LoadConfig(cfgPath)
-		if err != nil {
-			panic(fmt.Sprintf("unable to load configuration file for private transaction manager from %s due to %s", cfgPath, err))
-		}
-		socketPath = filepath.Join(cfg.WorkDir, cfg.Socket)
-	}
 
 	client := &engine.Client{
 		HttpClient: &http.Client{
-			Transport: unixTransport(socketPath),
+			Transport: unixTransport(cfg),
 		},
 		BaseURL: "http+unix://c",
 	}
 	ptm, err := selectPrivateTxManager(client)
 	if err != nil {
-		panic(fmt.Sprintf("unable to connect to private tx manager using %s due to %s", socketPath, err))
+		panic(fmt.Sprintf("unable to connect to private tx manager using %s due to %s", cfgPath, err))
 	}
 	return ptm
 }
 
-func unixTransport(socketPath string) *httpunix.Transport {
+func unixTransport(cfg engine.Config) *httpunix.Transport {
 	t := &httpunix.Transport{
-		DialTimeout:           1 * time.Second,
-		RequestTimeout:        5 * time.Second,
-		ResponseHeaderTimeout: 5 * time.Second,
+		DialTimeout:           time.Duration(cfg.DialTimeout) * time.Second,
```

Comment:
I'm not entirely comfortable removing these in case it causes issues in existing uses. I'd prefer to just add the `http.Client` timeout as an additional config option.

Reply:
Having experimented with the use of the http.Client timeout, I don't think it's actually possible to use it.
Whilst the client.timeout does expire, the socket call still blocks until the responseHeader timeout expires within the Transport.